modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C

```diff
@@ -0,0 +1,204 @@
+
+  // Current ace normal
+  const auto _normal = _face_info->normal();
+  /// ---------------------------------------------------------
+
+  /// ------------------ Coupled variables ---------------------
+
+  // Computing the gradient from coupled variables
+  // Normally, we can do this with `_var.gradient(face)` but we will need the transpose gradient
+  // So, we compute all at once
+  ADRealTensorValue gradient;
+  if (_dim == 1)
+  {
+    //const auto & grad_u = _u_var->adGradSln(face_info, skewness_correction);
+    const auto & grad_u = _u_var->gradient(face);
+    gradient = ADRealTensorValue(grad_u);
+  }
+  else if (_dim == 2)
+  {
+    // const auto & grad_u = _u_var->adGradSln(face_info, skewness_correction);
+    // const auto & grad_v = _v_var->adGradSln(face_info, skewness_correction);
+    const auto & grad_u = _u_var->gradient(face);
+    const auto & grad_v = _v_var->gradient(face);
+    gradient = ADRealTensorValue(grad_u, grad_v, ADRealVectorValue(0,0,0));
+  }
+  else // if (_dim == 3)
+  {
+    // const auto & grad_u = _u_var->adGradSln(face_info, skewness_correction);
+    // const auto & grad_v = _v_var->adGradSln(face_info, skewness_correction);
+    // const auto & grad_w = _w_var->adGradSln(face_info, skewness_correction);
+    const auto & grad_u = _u_var->gradient(face);
+    const auto & grad_v = _v_var->gradient(face);
+    const auto & grad_w = _w_var->gradient(face);
+    gradient = ADRealTensorValue(grad_u, grad_v, grad_w);
+  }
+
+  // Getting transpose of the gradient matrix
+  auto gradient_transpose = gradient.transpose();
+
+  /// ---------------------------------------------------------
+
+  /// ------ Interpolating the diffusion coefficient with different methods ------
+
+  auto mu_elem = has_elem ? _mu(elem_face) : _mu(neighbor_face);
+  auto mu_neighbor = has_neighbor ? _mu(neighbor_face) : _mu(elem_face);
+
+  const Real distance_to_neighbor = _face_info->dCN().norm();
+  const Real distance_to_element   = _face_info->eCN().norm();
+  const Real total_distance        = distance_to_neighbor + distance_to_element;
+
+  // Ideally, we would do the following, but we optimize some operations:
+    //   const Real coef_elem = distance_to_neighbor / total_distance;
+    //   const Real coef_neighbor = distance_to_element / total_distance;
+
+    //   ADReal mu_face;
+    //   if (_harmonic_interpolation)
+    //     mu_face = 1.0 / (coef_neighbor / mu_neighbor + coef_elem / mu_elem);
+    //   else
+    //     mu_face = coef_neighbor * mu_neighbor + coef_elem * mu_elem;
+
+  const Real coef_elem = distance_to_neighbor / total_distance;
+
+  ADReal mu_face;
+  Real protection_mu = 1e-12;
+  if (_harmonic_interpolation)
+  {
+    const auto inv_mu_neighbor = 1.0 / (mu_neighbor + protection_mu);
+    mu_face = 1.0 / (coef_elem * (1/(mu_elem + protection_mu) - inv_mu_neighbor) + inv_mu_neighbor);
+  }
+  else
+    mu_face = coef_elem * (mu_elem - mu_neighbor) + mu_neighbor;
```

Comment:
You can actually use an interpolation parameter and collapse this if statement into one line. Take a look at `FVDiffusion`.  I tested and `coef_elem * (mu_elem - mu_neighbor) + mu_neighbor` is not faster than the one used in the interpolation function. I suspect the compiler takes care of that in the background.